### PR TITLE
Postgis enhancements

### PR DESCRIPTION
**Pull request by avancinirodrigo**
- [x] #2186 
- [x] #2031 
- [x] #2184 
- [x] #2222 
- [x] #2223 
- [x] #1785

### CODE REVIEW

File: packages/gis/lua/TerraLib.lua [hunk 1]
```diff
@@ -553,19 +553,31 @@ local function createCellSpaceLayer(inputLayer, name, dSetName, resolution, conn
...
+		errorMsg = cellSpaceOpts:createCellSpace(cellLayerInfo, cellName, resolution, resolution,
+												inputLayer:getExtent(), inputLayer:getSRID(), cLType)
 	end
 
-	cellSpaceOpts:createCellSpace(cellLayerInfo, cellName, resolution, resolution,
-								inputLayer:getExtent(), inputLayer:getSRID(), cLType)
+	if errorMsg ~= "" then
+		if (type == "POSTGIS") and (string.find(errorMsg, "invalid SRID")) then
+			customError("Layer '"..inputLayer:getTitle().."' has a invalid EPSG. "
+						.. "Please, set a valid projection, if it still doesn't work, "
```
Comment (pedro-andrade-inpe) [inline packages/gis/lua/TerraLib.lua]: "[...] has AN invalid EPSG. Please select a valid projection or reproject the data."

File: packages/gis/lua/Layer.lua [hunk 2]
```diff
@@ -483,6 +483,16 @@ Layer_ = {
 	delete = function(self)
 		deleteData(self)
 	end,
+	--- Drop a database.
```
Comment (pedro-andrade-inpe) [inline packages/gis/lua/Layer.lua]: Database or layer?

"This function only works when the layer is stored in a PostGIS table."

File: packages/gis/lua/Layer.lua [hunk 3]
```diff
@@ -483,6 +483,16 @@ Layer_ = {
...
+	-- @usage -- DONTRUN
+	-- layer:drop()
+	drop = function(self)
+		if self.database then
+			TerraLib().dropPgDatabase(self)
+		else
+			customError("Function 'drop' only works with PostGIS database.")
```
Comment (pedro-andrade-inpe) [inline packages/gis/lua/Layer.lua]: PostGIS layer?

File: packages/gis/tests/postgis/alternative/Layer.lua [hunk 4]
```diff
@@ -291,28 +281,7 @@ return {
...
-		else
-			unitTest:assertError(portWrong, "It was not possible to create a connection to the given data source due to the following error: ".. -- SKIP
-				"could not connect to server: Connection refused (0x0000274D/10061)\n"..
-				"\tIs the server running on host \"localhost\" (::1) and accepting\n"..
-				"\tTCP/IP connections on port 2345?\n"..
-				"could not connect to server: Connection refused (0x0000274D/10061)\n"..
-				"\tIs the server running on host \"localhost\" (127.0.0.1) and accepting\n"..
-				"\tTCP/IP connections on port 2345?\n.")
-		end
+		unitTest:assertError(portWrong,"Port '2345' is not configured to host 'localhost'.")
```
Comment (pedro-andrade-inpe) [inline packages/gis/tests/postgis/alternative/Layer.lua]: [...] is not available for host [...]

File: packages/gis/tests/postgis/alternative/Layer.lua [hunk 5]
```diff
@@ -329,14 +298,7 @@ return {
...
 
-		if sessionInfo().system == "linux" then
-			unitTest:assertError(userNotExists, "It was not possible to create a connection to the given data source due to the following error: ".. -- SKIP
-				"FATAL:  password authentication failed for user \"usernotexists\"\n"..
-				"FATAL:  password authentication failed for user \"usernotexists\"\n.")
-		else
-			unitTest:assertError(userNotExists, "It was not possible to create a connection to the given data source due to the following error: ".. -- SKIP
-				"FATAL:  password authentication failed for user \"usernotexists\"\n.")
-		end
+		unitTest:assertError(userNotExists, "Connection failed, invalid username or password.")
```
Comment (pedro-andrade-inpe) [inline packages/gis/tests/postgis/alternative/Layer.lua]: [...] failed due to invalid [...]

File: packages/gis/lua/Layer.lua (hunk 2 above)
Comment (avancinirodrigo) [reply packages/gis/lua/Layer.lua]: This function works like 'delete' function. It doesn't drop the layer, it just remove the database.

File: packages/gis/lua/Layer.lua (hunk 3 above)
Comment (avancinirodrigo) [reply packages/gis/lua/Layer.lua]: Ok

File: packages/gis/tests/postgis/alternative/Layer.lua (hunk 5 above)
Comment (avancinirodrigo) [reply packages/gis/tests/postgis/alternative/Layer.lua]: Ok

File: packages/gis/lua/Layer.lua (hunk 2 above)
Comment (pedro-andrade-inpe) [reply packages/gis/lua/Layer.lua]: "Note that it removes all tables within the database."

**Pull request merged**